src/raftstore/store/fsm/peer.rs

```diff
@@ -1489,10 +1534,23 @@ impl<T: Transport, C: PdClient> Store<T, C> {
 
         // Check whether the store has the right peer to handle the request.
         let peer = self.check_propose_peer(msg)?;
```

Comment:
How about inline the `check_propose_peer`? So that we borrow the `region_peers` instead of the `self`.

Reply:
```rust
let region_id = msg.get_header().get_region_id();
        let peer = match self.region_peers.get(&region_id) {
            Some(peer) => peer,
            None => return Err(Error::RegionNotFound(region_id)),
        };
        if !peer.is_leader() {
            return Err(Error::NotLeader(
                region_id,
                peer.get_peer_from_cache(peer.leader_id()),
            ));
        }
```
Your saying to move this block to this fucntion right?

Reply:
Yes, it looks like resolve borrow issue.